agent/xds/clusters.go

```diff
@@ -849,6 +849,20 @@ func (s *ResourceGenerator) configIngressUpstreamCluster(c *envoy_cluster_v3.Clu
 	if threshold != nil {
 		c.CircuitBreakers.Thresholds = []*envoy_cluster_v3.CircuitBreakers_Thresholds{threshold}
 	}
+
+	// Configure the outlier detector for upstream service
+	var override *structs.PassiveHealthCheck
+	if svc != nil {
+		override = svc.PassiveHealthCheck
+	}
+	outlierDetection := ToOutlierDetection(cfgSnap.IngressGateway.Defaults.PassiveHealthCheck, override)
+
+	// Specail handling for failover peering service, which has set MaxEjectionPercent
+	if c.OutlierDetection != nil && c.OutlierDetection.MaxEjectionPercent != nil {
```

Comment:
Sorry for yet another comment. I just noticed that we aren't setting panic threshold here [similar to other clusters](https://github.com/hashicorp/consul/blob/main/agent/xds/clusters.go#L264-L266). For the sake of consistency, I'm guessing we want to set it like we do on other cluster types (because panic threshold interacts with outlier detection).

Reply:
Sorry, I am little confused since this has been set for the upstream clusters of the ingress gateway in https://github.com/hashicorp/consul/blob/cd53120cd7496c21892494c27f234ef1d3623081/agent/xds/clusters.go#L1147, which is called in making upstream clusters for the ingress-gateway service https://github.com/hashicorp/consul/blob/cd53120cd7496c21892494c27f234ef1d3623081/agent/xds/clusters.go#L789-L796